### PR TITLE
Benchmark sad defunding of a ledger-funded channel

### PR DESCRIPTION
**Pull request by geoknee**
Follows the established benchmarking patterns. introduces a ledger channel funding the channel in question. Both are challenged and the money is pushed down from L to X and then out. (The scenario being that the counter-party goes offline).

### CODE REVIEW

File: packages/nitro-protocol/gas-benchmarks/gas.ts [hunk 1]
```diff
@@ -62,11 +65,23 @@ export const gasRequiredTo: GasRequiredTo = {
...
     },
   },
+  ETHexitSadLedgerFunded: {
+    // Scenario: counterparty goes offline
+    vanillaNitro: {
+      challengeX: 93404,
+      challengeL: 92122,
+      pushOutcomeAndTransferAllL: 58640,
+      pushOutcomeAndTransferAllX: 107742,
+      total: 351908, // An alternative would be to 1. pushOutcome then 2. transfer(indices)
```
Comment (geoknee) [inline packages/nitro-protocol/gas-benchmarks/gas.ts:83]: TLDR it costs 352K gas to exit a ledger-funded channel in the worst case.

File: packages/nitro-protocol/gas-benchmarks/benchmark.test.ts [hunk 2]
```diff
@@ -105,36 +111,27 @@ describe('Consumes the expected gas for happy-path exits', () => {
...
     await (await ethAssetHolder.deposit(channelId, 0, 10, {value: 10})).wait();
-    const finalizesAtPromise = new Promise(resolve =>
-      nitroAdjudicator.once('ChallengeRegistered', (...event) => {
-        resolve(event[2]); // 2 is the position of finalizesAt
-      })
-    );
-    async function waitForChallengeToTimeOut(finalizesAt) {
-      await provider.send('evm_setNextBlockTimestamp', [finalizesAt + 1]);
-      await provider.send('evm_mine', []);
-    }
```
Comment (NiloCK) [inline packages/nitro-protocol/gas-benchmarks/benchmark.test.ts:121]: Glad to see this moved out. I should have commented on it in the first place.

File: packages/nitro-protocol/gas-benchmarks/gas.ts [hunk 3]
```diff
@@ -17,7 +17,12 @@ type Path =
   | 'ETHexitSadLedgerFunded';
 
 // The channel being benchmarked is a 2 party null app funded with 5 wei / tokens each.
-
+// KEY
```
Comment (NiloCK) [inline packages/nitro-protocol/gas-benchmarks/gas.ts:20]: These diagrams are straightforward but quite useful.

Should each step be included inside the relevant implementing test at the appropriate spots?

**Review by NiloCK — APPROVED** [review]
This looks good to me, although I confess it took me a few minutes to parse the ledger funded sad exit.

I pushed a trivial commit on top of these (separating each test case with a blank line), which seemed like an appropriate thing to instead of opening 5-10 suggested edits.

File: packages/nitro-protocol/gas-benchmarks/gas.ts (hunk 3 above)
Comment (geoknee) [reply packages/nitro-protocol/gas-benchmarks/gas.ts:20]: I've got some improvements coming to these diagrams in a stacked PR. I think putting them in the implementation is also a good idea -- I'll take a look at that in a follow-up PR.

**Pull request merged**